src/pages/index.js

```diff
@@ -53,67 +54,206 @@ newCardPopup.setEventListeners();
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLikeClick,
+    handleUnlikeClick
+  );
   return card.getView();
 }
 
-//render card
 function renderCard(cardData) {
   const cardView = createCard(cardData);
+  const section = new Section({}, "#cards__list");
   section.addItem(cardView);
 }
 
-// declare a function to handle image click
 function handleImageClick({ name, link }) {
+  // open the popup with the image
   popupWithImage.open({ link, name });
 }
 
-// EVENT HANDLERS
+// =========  EVENT HANDLERS  ========= //
 
 // Handles edit form submit
-function handleProfileEditSubmit() {
-  editFormValidator.disableSubmitButton();
-  editProfilePopup.close();
+function handleProfileEditSubmit({ title, description }) {
+  api
+    .editProfileInfo({
+      name: title,
+      about: description,
+    })
+    .then((data) => {
+      userInfo.setUserInfo(data.name, data.about);
+      editFormValidator.disableSubmitButton();
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 
+// handles new card submission
 function handleNewCardSubmit(name, link) {
-  renderCard({ name, link }, placesWrap);
-  cardFormValidator.disableSubmitButton();
-  newCardPopup.close();
+  api
+    .addNewCard({
+      name: name,
+      link: link,
+    })
+    .then((data) => {
+      renderCard(data);
+      cardFormValidator.disableSubmitButton();
+      newCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+// handles deleting card
+function handleDeleteCard(card) {
+  // open the delete card popup
+  deleteCardPopup.open();
+  deleteCardPopup.setSubmitCallback(() => {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.deleteCard();
+        deleteCardPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+// function to handle like button click
+function handleLikeClick(card) {
+  // const cardId = card.getId();
+  api
+    .likeCard(card._id)
+    .then((data) => {
+      card._handleLikeIcon();
+      card._isLiked = true;
+      console.log("Card liked:", data);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 
-// EVENT LISTENERS
+function handleUnlikeClick(card) {
+  api
+    .removeLike(card._id)
+    .then((data) => {
+      card._handleLikeIcon();
+      card._isLiked = false;
+      console.log("Card disliked:", data);
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+}
+
+// handles avatar form submit with api
+function handleAvatarSubmit() {
+  const avatar = document.querySelector("#avatar-link-input").value;
+  api
+    .updateAvatar({
+      avatar: avatar,
+    })
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      avatarFormValidator.disableSubmitButton();
+      changeAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors